bentoml/adapters/annotated_image_input.py

```diff
@@ -0,0 +1,372 @@
+# Copyright 2019 Atalaya Tech, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import json
+import os
+import argparse
+from io import BytesIO
+from typing import Iterable
+
+from werkzeug.utils import secure_filename
+from werkzeug.wrappers import Request
+
+from bentoml import config
+from bentoml.utils.lazy_loader import LazyLoader
+from bentoml.marshal.utils import SimpleRequest, SimpleResponse
+from bentoml.exceptions import BadInput
+from bentoml.adapters.base_input import BaseInputAdapter
+
+# BentoML optional dependencies, using lazy load to avoid ImportError
+imageio = LazyLoader('imageio', globals(), 'imageio')
+
+
+def has_json_extension(file_name):
+    """
+    Check if file's extension is an acceptable JSON extension
+    (currently only .json is allowed)
+    """
+    _, extension = os.path.splitext(file_name)
+
+    if extension.lower() in [".json"]:
+        return True
+
+    return False
+
+
+def has_image_extension(file_name, accept_format_list):
```

Comment:
I think that's a good idea, I'll get on that tomorrow.

Reply:
Alright!  I've added type hints for these in my latest commit.  I've also made `accept_format_list` required for both the `has_image_extension` and `verify_image_format_or_raise` functions.  If unset during `__init__`, it is set to the default value.  If the downstream code manages to change it to "None" sometime after object construction, the code will crash when trying to check image types (rather than silently defaulting back the default values).

This seemed reasonable to me, as messing with the accept_image_format_list after construction is unusual, and the caller should explicitly specify if they want to use the defaults.  It isn't clear (in my mind) whether `None` should imply "use the default" or should imply "an empty list where no images are accepted".

If you'd rather I changed it so that it gracefully interprets "None" as "use the defaults", that's no problem at all.